src/Npgsql/Internal/TypeHandlers/JsonHandler.cs

```diff
@@ -215,16 +215,17 @@ public override ValueTask<string> Read(NpgsqlReadBuffer buf, int len, bool async
         => ReadCustom<string>(buf, len, async, fieldDescription);
 
     /// <inheritdoc />
-    public TextReader GetTextReader(Stream stream)
+    public TextReader GetTextReader(Stream stream, int byteLength, NpgsqlReadBuffer buffer)
     {
         if (_isJsonb)
         {
             var version = stream.ReadByte();
+            byteLength -= 1;
```

Comment:
nit: can just do -1 when passing the argument below (not important)

Reply:
Yes, but that would require yet another check whether it's jsonb (so we have to reduce by one).

Reply:
Either way is fine for me. Maybe this shortcut makes sense for all the other handlers, but not for JsonHandler..